src/main/java/com/acme/edu/logger/command/CommandMessageString.java

```diff
@@ -31,6 +34,7 @@ public void typeSwitchUpdate(ControllerState state) {
         if (!state.getPreviousString().equals("")) {
```

Comment:
state.getPreviousString() can produce NPE. Please make check this way: "".equals(state.gPS())